torch/_dynamo/eval_frame.py

```diff
@@ -691,138 +696,174 @@ def result_capturing_wrapper(*graph_inputs):
         return result_capturing_wrapper
 
     flat_args, in_spec = pytree.tree_flatten((args, kwargs))
+    # apply constraints to flat_args
+    # First, reconstruct constraints to be a map:[int] view
+    if not constraints:
+        constraints = list()
 
-    remove_from_cache(f)
-    with patch(f"{__name__}.most_recent_backend", None), config.patch(
-        specialize_int=True
-    ):
-        opt_f = optimize_assert(
-            dynamo_normalization_capturing_compiler,
-            hooks=Hooks(guard_export_fn=guard_export_print, guard_fail_fn=None),
-            export=True,
-            dynamic=(tracing_mode == "symbolic"),
-        )(f)
-        # TODO(voz): We may have instances of `f` that mutate inputs, we should track sideffects and reject.
-        result_traced = opt_f(*args, **kwargs)
-    remove_from_cache(f)
+    tensor_to_indices: Dict[torch.Tensor, Set[int]] = dict()
+    try:
+        for constraint in constraints:
+            tensor = constraint[0]
+            index = constraint[1]
+            # TODO(voz): Is tensor defaultdict friendly?
```

Comment:
Use WeakIdKeyDictionary from torch.utils.weak, which does proper id semantics even though Tensor defines value equality for ==.

Reply:
Or just map from id to int, this dict is very short lived.